lib/prawn/graphics/color.rb

```diff
@@ -94,7 +94,7 @@ def process_color(*color)
       def color_type(color)
         case color
         when String
-          if color =~ /\A[0-F]{6}\z/i
+          if color =~ /\A[0-9|A-F]{6}\z/i
```

Comment:
By adding a pipe here, you are allowing the pipe character too. The correct expression is:

```
/\A[0-9A-F]{6}\z/i
```

You can test this by running the following in irb:

```
/[0-9|A-F]/.match('|').inspect
```

vs

```
/[0-9A-F]/.match('|').inspect
```